### PR TITLE
[MC][AMDGPU] Support .reloc BFD_RELOC_{NONE,32,64}

### PR DESCRIPTION
**Pull request by EthanLuisMcDonough**
Emitting BFD_RELOC_* reloc directives can cause internal errors on AMDGPU.

### CODE REVIEW

**Review by jhuber6 — COMMENTED** [review]
Needs a test

**Review by jhuber6 — COMMENTED** [review]
If we actually can handle these it would go in the AMDGPU relocations instead of being handled specially here. I don't know what these relocations are, do you know where they come from?

File: llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUAsmBackend.cpp [hunk 1]
```diff
@@ -163,12 +163,17 @@ void AMDGPUAsmBackend::applyFixup(const MCAssembler &Asm, const MCFixup &Fixup,
...
-#define ELF_RELOC(Name, Value)                                                 \
-  .Case(#Name, MCFixupKind(FirstLiteralRelocationKind + Value))
+  auto Type = StringSwitch<unsigned>(Name)
+#define ELF_RELOC(Name, Value) .Case(#Name, Value)
 #include "llvm/BinaryFormat/ELFRelocs/AMDGPU.def"
 #undef ELF_RELOC
-      .Default(std::nullopt);
+                  .Case("BFD_RELOC_NONE", ELF::R_AMDGPU_NONE)
+                  .Case("BFD_RELOC_32", ELF::R_AMDGPU_ABS32)
+                  .Case("BFD_RELOC_64", ELF::R_AMDGPU_ABS64)
```
Comment (jdoerfert) [inline llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUAsmBackend.cpp:172]: Why are they not added to "llvm/BinaryFormat/ELFRelocs/AMDGPU.def" ?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUAsmBackend.cpp:172]: I have no idea why it's like this, why wouldn't these all be in the def? I guess it's fine to copy the pattern and clean them all later?

Comment (jdoerfert) [reply llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUAsmBackend.cpp:172]: We just need to decide which way we want to do it.

**Review by arsenm — APPROVED** [review]
Follow up could move all the targets to use the def files

**Pull request merged**